blog/2023-3-5-magazine-002.md

```diff
@@ -0,0 +1,249 @@
+
+In the following example, we are declaring a resource called `DoubleBucket`. When an object is `put()` into the resource
+it will be replicated into two buckets.
+
+```js
+bring cloud;
+
+resource DoubleBucket {
+  a: cloud.Bucket;
+  b: cloud.Bucket;
+  
+  init(a: cloud.Bucket, b: cloud.Bucket) {
+    this.a = a;
+    this.b = b;
+  }
+  
+  inflight put(key: str, obj: str) {
+    this.a.put(key, obj);
+    this.b.put(key, obj);
+  }
+  
+  inflight get(key: str): str {
+    return this.a.get(key);
+  }
+}
+
+let left = new cloud.Bucket();
+let right = new cloud.Bucket();
+
+let r = new DoubleBucket(left, right);
+
+new cloud.Function(inflight (s: str): str => {
+  r.put("hello.txt", "world");
+});
+```
+
+Within that resource, you can see the concept of an `inflight` modifier on both the `put()` and `get()` methods. If you need a reminder of the terms inflight and preflight, read the [spec](https://docs.winglang.io/reference/spec#13-phase-modifiers) or the [inflight functions](https://docs.winglang.io/concepts/inflights) section in our docs.
+
+This example demonstrates how user-defined resources can create composable units that allow for the creation of higher-level abstractions.
+
+There are many capabilities that user-defined resources open up, and you can read more about them in the [language specification](https://docs.winglang.io/reference/spec#33-resources), as well as in specific issues and [pull requests](https://github.com/winglang/wing/pull/1432) related to the feature. The separation of the inflight and preflight scopes is a particularly interesting aspect of this feature, and there is ongoing work to refine and optimize this separation.
+
+If you would like to see it in action, Yoav Steinberg created a video that demonstrates how to create user resources:
+
+<iframe 
+  width="560" 
+  height="315" 
+  src="https://www.youtube.com/watch?v=eAjQU5VUJpg" 
+  title="Wing Console support for filtering logs" 
+  frameborder="0"
+  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" 
+  allowfullscreen>
+</iframe>
+
+### Introducing new types to the language 
+
+We are still enriching the [Wing Language Specification](https://docs.winglang.io/reference/spec) and increasing the set of features we believe are required to build cloud applications with Wing.
+
+We added support for `for..in` ([#1412](https://github.com/winglang/wing/pull/1412/)) as well as the `duration` and `datetime` types ([#1470](https://github.com/winglang/wing/pull/1470)). There are also duration literals minutes (e.g `5m`), seconds (e.g. `10s`), and hours (e.g. `5h`).
+
+Let's take a look at a quick example:
```

Comment:
Maybe also add an example of the `duration` type:

```js
let timeout = 5m;
assert(timeout.seconds == 300);
```

Reply:
Done.